packages/core/src/transport/httpRequest.ts

```diff
@@ -13,26 +15,55 @@ import { addTelemetryError } from '../domain/telemetry'
 export type HttpRequest = ReturnType<typeof createHttpRequest>
 
 export function createHttpRequest(endpointBuilder: EndpointBuilder, bytesLimit: number) {
-  return {
-    send: (data: string | FormData, bytesCount: number) => {
-      const url = endpointBuilder.build()
-      const canUseBeacon = !!navigator.sendBeacon && bytesCount < bytesLimit
-      if (canUseBeacon) {
-        try {
-          const isQueued = navigator.sendBeacon(url, data)
-
-          if (isQueued) {
-            return
-          }
-        } catch (e) {
-          reportBeaconError(e)
+  function sendBeacon(data: string | FormData, bytesCount: number) {
+    const url = endpointBuilder.build()
+    const canUseBeacon = !!navigator.sendBeacon && bytesCount < bytesLimit
+    if (canUseBeacon) {
+      try {
+        const isQueued = navigator.sendBeacon(url, data)
+
+        if (isQueued) {
+          return
         }
+      } catch (e) {
+        reportBeaconError(e)
       }
+    }
+
+    sendXHR(url, data)
+  }
+
+  function sendFetch(data: string | FormData, bytesCount: number) {
+    const url = endpointBuilder.build()
+    const canUseKeepAlive = window.Request && 'keepalive' in new Request('') && bytesCount < bytesLimit
+    if (canUseKeepAlive) {
+      fetch(url, { method: 'POST', body: data, keepalive: true }).catch(
+        monitor(() => {
+          // failed to queue the request
+          sendXHR(url, data)
+        })
+      )
+    } else {
+      sendXHR(url, data)
+    }
+  }
+
+  function sendXHR(url: string, data: string | FormData) {
+    const request = new XMLHttpRequest()
+    request.open('POST', url, true)
+    request.send(data)
+  }
 
-      const request = new XMLHttpRequest()
-      request.open('POST', url, true)
-      request.send(data)
+  return {
+    send: (data: string | FormData, bytesCount: number) => {
+      // TODO check tests without flag
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏not sure what this comment means?

Reply:
just a leftover :swe